src/pip/_internal/cli/base_command.py

```diff
@@ -87,6 +87,13 @@ def __init__(self, name, summary, isolated=False):
             self.parser,
         )
         self.parser.add_option_group(gen_opts)
+        try:
+            # mypy raises error due to
+            # https://github.com/python/mypy/issues/5868
+            self.add_options(self.cmd_opts)  # type: ignore
+        except AttributeError:
+            # it means that the base class has not defined the method
+            logger.debug("No add_options method defined in the class")
```

Comment:
Is this debug message adding anything useful? Also this message will show up in the logs when a command is run with verbose flags to an end user, and potentially be confusing, since it's not directly related to the command itself.

I suggest removing this, and adding `pass`. The comment can be moved before `try` to explain why we are catching `AttributeError`

Reply:
@deveshks using an empty `add_options` with only `pass` statement is what I used earlier. However, it was providing conflicts in some tests where `Command` class was being used directly and since `Command` class does not define this method, rather its child classes, the tests were failing.

Another reason for using it that way is that, what I felt, is being repetitive and restrictive if we create other command classes which do not really define its own `cmd_opts`. They would still have to create an empty method.

I am sure this is something where suggestions are always welcome. Let's see what others have to say about this.

Reply:
Agreed, but we can always update the test classes. 

Also my comment was more in line with what was said here https://github.com/pypa/pip/issues/8244#issuecomment-631538850
, especially the line `with the default implementation doing nothing.` which tells me to create a default method in base class with pass (Unless I am interpreting it incorrectly)